<?xml version="1.0"?>

<launch>
  <arg name="namespace" default="/vigir/footstep_planning" />
  <group ns="$(arg namespace)">
    <!-- run map servers -->
    <node name="map_server_1" pkg="map_server" type="map_server" args="$(find vigir_footstep_planner)/maps/RCGO2011_Hector_Mission4_2.yaml">
      <param name="frame_id" value="/world" />
      <remap from="map" to="/body_level_grid_map" />
    </node>

    <node name="map_server_2" pkg="map_server" type="map_server" args="$(find vigir_footstep_planner)/maps/RCGO2011_Hector_Mission4_2.yaml">
      <param name="frame_id" value="/world" />
      <remap from="map" to="/ground_level_grid_map" />
    </node>

    <!-- start up footstep planning system -->
    <include file="$(find vigir_footstep_planning)/launch/footstep_planning_basics.launch">
      <arg name="namespace" value="$(arg namespace)" />
    </include>

    <!-- start footstep planner -->
    <!-- launch-prefix="valgrind -\-tool=callgrind -\-log-file=/home/alex/vigir/cachegrind/cachegrind.out" -->
    <node name="footstep_planner" pkg="vigir_footstep_planner" type="footstep_planner_node" respawn="true" output="screen">
      <remap from="body_level_grid_map" to="ground_level_grid_map" />
    </node>

    <!-- load plugin configurations from YAML file to parameter server -->
    <rosparam file="$(find vigir_footstep_planner)/config/default_plugin_sets.yaml" command="load" />

    <!-- specify where to load robot specific parameters -->
    <rosparam file="$(find vigir_footstep_planner)/config/robot_params.yaml" command="load" />
    <param name="planner_configs_path" value="$(find vigir_footstep_planner)/config/planner/" />
  </group>
</launch>
